### ISSUE TITLE
conda init resolves to real path instead of absolute path

### ISSUE THREAD

**Issue by Hemario** [conda/conda#13302]
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

We have a setup where we install newer versions of miniforge into the same folder and symbolic link to latest

example:
```
$ ls -ln /opt/anaconda/miniforge3/
drwxrwxr-x. 19 1000309 1000309 4096 Nov  8 11:04 23.3.1-1
drwxrwxr-x. 19 1000309 1000309 4096 Nov  8 12:08 23.3.1-1-test
lrwxrwxrwx.  1 1000309 1000309   33 Oct 27 15:27 latest -> /opt/anaconda/miniforge3/23.3.1-1
```

However, if I run `conda init` or `mamba init` from the latest folder, the .bashrc file gets populated with the real path (containing the version number) instead of the relative path (with latest), making the latest softlink kinda obsolete.

Is this working as intended?

```
$ /opt/anaconda/miniforge3/latest/bin/conda init
```
contents of `.bashrc`:
```
# >>> conda initialize >>>
# !! Contents within this block are managed by 'conda init' !!
__conda_setup="$('/opt/anaconda/miniforge3/23.3.1-1/bin/conda' 'shell.bash' 'hook' 2> /dev/null)"
if [ $? -eq 0 ]; then
    eval "$__conda_setup"
else
    if [ -f "/opt/anaconda/miniforge3/23.3.1-1/etc/profile.d/conda.sh" ]; then
        . "/opt/anaconda/miniforge3/23.3.1-1/etc/profile.d/conda.sh"
    else
        export PATH="/opt/anaconda/miniforge3/23.3.1-1/bin:$PATH"
    fi
fi
unset __conda_setup
# <<< conda initialize <<<
```

### Conda Info

```shell
active environment : base
    active env location : /opt/anaconda/miniforge3/23.3.1-1
            shell level : 1
       user config file : /home/AD/xxxxxxx/.condarc
 populated config files : /opt/anaconda/miniforge3/23.3.1-1/.condarc
                          /home/AD/xxxxxxx/.conda/.condarc
          conda version : 23.3.1
    conda-build version : not installed
         python version : 3.10.12.final.0
       virtual packages : __archspec=1=x86_64
                          __glibc=2.28=0
                          __linux=4.18.0=0
                          __unix=0=0
       base environment : /opt/anaconda/miniforge3/23.3.1-1  (read only)
      conda av data dir : /opt/anaconda/miniforge3/23.3.1-1/etc/conda
  conda av metadata url : None
           channel URLs : our-artifactory-channels
          package cache : /home/AD/xxxxxxx/.conda/pkgs
       envs directories : /home/AD/xxxxxxx/.conda/envs
                          /opt/anaconda/miniforge3/23.3.1-1/envs
               platform : linux-64
             user-agent : conda/23.3.1 requests/2.31.0 CPython/3.10.12 Linux/4.18.0-477.15.1.el8_8.x86_64 rhel/8.8 glibc/2.28
                UID:GID : 2244429:10000000
             netrc file : None
           offline mode : False
```

### Conda Config

```shell
==> /opt/anaconda/miniforge3/23.3.1-1/.condarc <==
channels:
  - conda-forge

==> /home/AD/xxxxxxx/.conda/.condarc <==
pkgs_dirs:
  - $HOME/.conda/pkgs
channels:
  our-artifactory-channels
```

### Conda list

```shell
# packages in environment at /opt/anaconda/miniforge3/23.3.1-1:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
boltons                   23.0.0             pyhd8ed1ab_0    conda-forge
brotli-python             1.1.0           py310hc6cd4ac_0    conda-forge
bzip2                     1.0.8                h7f98852_4    conda-forge
c-ares                    1.19.1               hd590300_0    conda-forge
ca-certificates           2023.7.22            hbcca054_0    conda-forge
certifi                   2023.7.22          pyhd8ed1ab_0    conda-forge
cffi                      1.15.1          py310h255011f_3    conda-forge
charset-normalizer        3.2.0              pyhd8ed1ab_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
conda                     23.3.1          py310hff52083_0    conda-forge
conda-libmamba-solver     23.3.0             pyhd8ed1ab_0    conda-forge
conda-package-handling    2.2.0              pyh38be061_0    conda-forge
conda-package-streaming   0.9.0              pyhd8ed1ab_0    conda-forge
cryptography              41.0.3          py310h75e40e8_0    conda-forge
fmt                       9.1.0                h924138e_0    conda-forge
icu                       73.2                 h59595ed_0    conda-forge
idna                      3.4                pyhd8ed1ab_0    conda-forge
jsonpatch                 1.32               pyhd8ed1ab_0    conda-forge
jsonpointer               2.0                        py_0    conda-forge
keyutils                  1.6.1                h166bdaf_0    conda-forge
krb5                      1.21.2               h659d440_0    conda-forge
ld_impl_linux-64          2.40                 h41732ed_0    conda-forge
libarchive                3.6.2                h039dbb9_1    conda-forge
libcurl                   8.2.1                hca28451_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 h516909a_1    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 13.1.0               he5830b7_0    conda-forge
libgomp                   13.1.0               he5830b7_0    conda-forge
libiconv                  1.17                 h166bdaf_0    conda-forge
libmamba                  1.4.2                hcea66bb_0    conda-forge
libmambapy                1.4.2           py310h1428755_0    conda-forge
libnghttp2                1.52.0               h61bc06f_0    conda-forge
libnsl                    2.0.0                h7f98852_0    conda-forge
libsolv                   0.7.24               hfc55251_3    conda-forge
libsqlite                 3.43.0               h2797004_0    conda-forge
libssh2                   1.11.0               h0841786_0    conda-forge
libstdcxx-ng              13.1.0               hfd8a6a1_0    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libxml2                   2.11.5               h232c23b_1    conda-forge
libzlib                   1.2.13               hd590300_5    conda-forge
lz4-c                     1.9.4                hcb278e6_0    conda-forge
lzo                       2.10              h516909a_1000    conda-forge
mamba                     1.4.2           py310h51d5547_0    conda-forge
ncurses                   6.4                  hcb278e6_0    conda-forge
openssl                   3.1.2                hd590300_0    conda-forge
packaging                 23.1               pyhd8ed1ab_0    conda-forge
pip                       23.2.1             pyhd8ed1ab_0    conda-forge
pluggy                    1.3.0              pyhd8ed1ab_0    conda-forge
pybind11-abi              4                    hd8ed1ab_3    conda-forge
pycosat                   0.6.4           py310h5764c6d_1    conda-forge
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pyopenssl                 23.2.0             pyhd8ed1ab_1    conda-forge
pysocks                   1.7.1              pyha2e5f31_6    conda-forge
python                    3.10.12         hd12c33a_0_cpython    conda-forge
python_abi                3.10                    3_cp310    conda-forge
readline                  8.2                  h8228510_1    conda-forge
reproc                    14.2.4               h0b41bf4_0    conda-forge
reproc-cpp                14.2.4               hcb278e6_0    conda-forge
requests                  2.31.0             pyhd8ed1ab_0    conda-forge
ruamel.yaml               0.17.32         py310h2372a71_0    conda-forge
ruamel.yaml.clib          0.2.7           py310h1fa729e_1    conda-forge
setuptools                68.1.2             pyhd8ed1ab_0    conda-forge
tk                        8.6.12               h27826a3_0    conda-forge
toolz                     0.12.0             pyhd8ed1ab_0    conda-forge
tqdm                      4.66.1             pyhd8ed1ab_0    conda-forge
tzdata                    2023c                h71feb2d_0    conda-forge
urllib3                   2.0.4              pyhd8ed1ab_0    conda-forge
wheel                     0.41.2             pyhd8ed1ab_0    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
yaml-cpp                  0.7.0                h27087fc_2    conda-forge
zstandard                 0.19.0          py310h5764c6d_0    conda-forge
zstd                      1.5.5                hfc55251_0    conda-forge
```

**Comment by travishathaway** [comment]
Hi @Hemario,

Yes, this is the expected behavior. The path that you are seeing is based on the `sys.prefix` value for the underlying Python installation (https://docs.python.org/3/library/sys.html?highlight=sys%20prefix#sys.prefix).

This is a fixed value and is coupled to where Python was initially installed to on your system.

I'm going to switch this over to a feature request instead of a bug.

I can't help but be a little curious about your setup though. What is motivating you to want to have multiple conda versions installed simultaneously?  Perhaps there's a better way to work around this.

**Comment by Hemario** [comment]
@travishathaway 
> I can't help but be a little curious about your setup though. What is motivating you to want to have multiple conda versions installed simultaneously? Perhaps there's a better way to work around this.

In our development and production environment we rely on conda/mamba from miniconda or miniforge for our internal deployment scripts (MLOps) and for setting up project environments for our data scientists. We want to be able to fall back to a previous version of conda/mamba in case we encounter problems with a new installation, by just updating our symbolic link.

Is this overkill? Probably. As usually these packages are tested thoroughly before release. However, I can't help but to be cautious.

**Comment by travishathaway** [comment]
@Hemario,

That's a valid use case and something that conda admittedly conda doesn't support too well right now.

I would suggest taking a look into something like [micromamba](https://mamba.readthedocs.io/en/latest/installation/micromamba-installation.html) as this is distributed as a single compiled binary. Making backups and creating symbolic links of the executable should be easier.

There have been experiments circulating for some time to create a single binary distribution of the conda command (see: https://github.com/conda/conda-standalone), but this is not ready for production usage yet. Once this is ready, it could in theory be a viable alternative to use.

Just as one more idea to throw out there for future work, we could also conceivably create a "conda version switcher" subcommand plugin. It could be a very simple command that functions like so:

```
conda version 23.9.1
```

If the specified version didn't currently exist, it would be downloaded and some things (not sure what yet) would happen to ensure that this is used as the current conda version.

**Comment by Hemario** [comment]
Thanks @travishathaway for the suggestions regarding `micromamba` and `conda-standalone`. We'll look into these to see if they are useful to us.

Currently we are thinking about a one-off script that would perform `sudo -u $username /opt/anaconda/miniforge-$version/bin/mamba init` for all users where we put $version to the stable version. That would mean we launch that one-off script every time we upgrade (or downgrade) to another version. Hopefully not that often.

Not entirely sure if I understand the conda version switcher and how that could work in our specific case as `conda init` generated code is placed in the individual user's .bashrc.
